src/bentoml/_internal/bento/build_config.py

```diff
@@ -544,15 +546,25 @@ def write_to_bento(self, bento_fs: FS, build_ctx: str) -> None:
             f.write(install_sh)
 
         if self.requirements_txt is not None:
-            requirements_txt_file = resolve_user_filepath(
-                self.requirements_txt, build_ctx
+            base_txt, nested_txts = find_nested_requirements_txts(
+                Path(
+                    resolve_user_filepath(self.requirements_txt, build_ctx),
+                )
             )
             copy_file_to_fs_folder(
-                requirements_txt_file,
+                str(base_txt.absolute()),
                 bento_fs,
                 py_folder,
                 dst_filename="requirements.txt",
             )
+            for txt in nested_txts:
+                dst_path = txt.relative_to(base_txt.parent)
```

Comment:
What does `dst_path` resolve to for `-r ../../requirements.txt`?

Reply:
It throws an exception. I should have noted somewhere that there is a requirement that nested files live at the same level or below the base file. To me this feels like a reasonable limitation.

Reply:
Yep, that makes sense to me. Can you catch that and throw a more user-friendly exception here that explains that we don't support it, then?

Reply:
Now that I've switched to using `pip-requirements-parser` as @aarnphm suggested this should no longer be a limitation as the parser is reading all nested files and essentially concating them all into a single file (with a small modification to make sure that we remove other file references).

I think the only thing which might not work not is `--find-links` with a path to an HTML file. However, that seems like a super niche edge case.